bazel/rules/golden_file_test.bzl

```diff
@@ -0,0 +1,51 @@
+"""Rule for testing generated files against golden files."""
+
+def _golden_file_test_impl(ctx):
+    """Implementation of golden_file_test rule."""
+
+    # Create a test script that diffs the files
+    test_script = ctx.actions.declare_file(ctx.label.name + "_test.sh")
+    ctx.actions.write(
+        output = test_script,
+        content = """#!/bin/bash -e
+if diff -u "{target}" "{golden}"; then
+    echo 'PASS: {name}'
+    exit 0
+else
+    echo 'FAIL: {name}'
+    echo '  To update the source do:'
+    echo '  cp "bazel-bin/{target}" "{golden}"'
```

Comment:
... and this one by [write_source_files](https://github.com/bazel-contrib/bazel-lib/blob/0a2f475200309e581f4e0d8aa6989eda7caed4e6/lib/write_source_files.bzl#L79-L84)?